```
Summarizing 1 Failure:
  [FAIL] Extension Required Runtime controller tests [It] should report dns extension as not required during garden deletion
  /home/prow/go/src/github.com/gardener/gardener/test/integration/operator/extension/required/runtime/runtime_test.go:307
Ran 8 of 9 Specs in 34.061 seconds
FAIL! -- 7 Passed | 1 Failed | 0 Pending | 1 Skipped
--- FAIL: TestExtensionRequiredRuntime (34.06s)
```

/test pull-gardener-integration

```
Summarizing 5 Failures:
  [INTERRUPTED] Shoot Tests Create Shoot, Rotate Credentials and Delete Shoot Shoot with workers with workers rollout [It] Wait for Shoot to be deleted [Shoot, default, credentials-rotation, basic, with-workers-rollout]
  /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/shoot/create_rotate_delete.go:342
  [INTERRUPTED] Shoot Tests Create Shoot, Rotate Credentials and Delete Shoot Shoot with workers with workers rollout, in-place update strategy [It] Wait for Shoot to be deleted [Shoot, default, credentials-rotation, basic, with-workers-rollout, in-place]
  /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/shoot/create_rotate_delete.go:342
  [TIMEDOUT] Shoot Tests Update from non-HA to HA with failure tolerance type 'node' Shoot with workers in-cluster access to API server [It] should access the API server via the kubernetes service's hostname [Shoot, high-availability, update-to-node]
  /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/shoot/internal/inclusterclient/inclusterclient.go:143
  [INTERRUPTED] Shoot Tests Create Shoot, Rotate Credentials and Delete Shoot Shoot with workers without workers rollout [It] Wait for Shoot to be reconciled [Shoot, default, credentials-rotation, basic, without-workers-rollout]
  /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/shoot/create_rotate_delete.go:231
  [INTERRUPTED] Shoot Tests Update from non-HA to HA with failure tolerance type 'node' Workerless Shoot [It] Wait for Shoot to be reconciled [Shoot, high-availability, update-to-node, workerless]
  /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/shoot/update-to-ha.go:33
Ran 285 of 833 Specs in 2004.540 seconds
FAIL! - Interrupted by Other Ginkgo Process -- 280 Passed | 5 Failed | 0 Pending | 548 Skipped
Ginkgo ran 1 suite in 34m2.76354665s
```

/test pull-gardener-e2e-kind-ha-single-zone

--- a/pkg/component/extensions/operatingsystemconfig/operatingsystemconfig_test.go
+++ b/pkg/component/extensions/operatingsystemconfig/operatingsystemconfig_test.go
@@ -56,7 +56,12 @@ import (
 
 var _ = Describe("OperatingSystemConfig", func() {
 	Describe("> Interface", func() {
-		const namespace = "test-namespace"
+		const (
+			namespace = "test-namespace"
+
+			worker1Name = "worker1"
+			worker2Name = "worker2"
+		)
 
 		var (
 			ctrl             *gomock.Controller
@@ -134,93 +139,15 @@ var _ = Describe("OperatingSystemConfig", func() {
 					nil
 			}
 
-			worker1Name = "worker1"
-			worker2Name = "worker2"
-			workers     = []gardencorev1beta1.Worker{
-				{
-					Name: worker1Name,
-					Machine: gardencorev1beta1.Machine{
-						Architecture: ptr.To(v1beta1constants.ArchitectureAMD64),
-						Image: &gardencorev1beta1.ShootMachineImage{
-							Name:           "type1",
-							Version:        ptr.To("12.34"),
-							ProviderConfig: &runtime.RawExtension{Raw: []byte(`{"foo":"bar"}`)},
-						},
-					},
-					KubeletDataVolumeName: &kubeletDataVolumeName,
-					ControlPlane:          &gardencorev1beta1.WorkerControlPlane{},
-				},
-				{
-					Name: worker2Name,
-					Machine: gardencorev1beta1.Machine{
-						Architecture: ptr.To(v1beta1constants.ArchitectureAMD64),
-						Image: &gardencorev1beta1.ShootMachineImage{
-							Name:    "type2",
-							Version: ptr.To("12.34"),
-						},
-					},
-					CRI: &gardencorev1beta1.CRI{
-						Name: gardencorev1beta1.CRINameContainerD,
-					},
-					KubeletDataVolumeName: &kubeletDataVolumeName,
-					Kubernetes: &gardencorev1beta1.WorkerKubernetes{
-						Version: &workerKubernetesVersion,
-					},
-				},
-			}
+			workers              []gardencorev1beta1.Worker
+			inPlaceUpdateWorkers []gardencorev1beta1.Worker
 
-			inPlaceUpdateWorkers = []gardencorev1beta1.Worker{
-				{
-					Name: worker1Name + "-in-place",
-					Machine: gardencorev1beta1.Machine{
-						Architecture: ptr.To(v1beta1constants.ArchitectureAMD64),
-						Image: &gardencorev1beta1.ShootMachineImage{
-							Name:           "type1",
-							Version:        ptr.To("12.34"),
-							ProviderConfig: &runtime.RawExtension{Raw: []byte(`{"foo":"bar"}`)},
-						},
-					},
-					KubeletDataVolumeName: &kubeletDataVolumeName,
-					Kubernetes: &gardencorev1beta1.WorkerKubernetes{
-						Kubelet: &gardencorev1beta1.KubeletConfig{
-							KubeReserved: &gardencorev1beta1.KubeletConfigReserved{
-								CPU:    ptr.To(resource.MustParse("100m")),
-								Memory: ptr.To(resource.MustParse("100Mi")),
-							},
-						},
-					},
-
-					UpdateStrategy: ptr.To(gardencorev1beta1.AutoInPlaceUpdate),
-				},
-				{
-					Name: worker2Name + "-in-place",
-					Machine: gardencorev1beta1.Machine{
-						Architecture: ptr.To(v1beta1constants.ArchitectureAMD64),
-						Image: &gardencorev1beta1.ShootMachineImage{
-							Name:    "type2",
-							Version: ptr.To("12.34"),
-						},
-					},
-					CRI: &gardencorev1beta1.CRI{
-						Name: gardencorev1beta1.CRINameContainerD,
-					},
-					KubeletDataVolumeName: &kubeletDataVolumeName,
-					Kubernetes: &gardencorev1beta1.WorkerKubernetes{
-						Version: &workerKubernetesVersion,
-						Kubelet: &gardencorev1beta1.KubeletConfig{},
-					},
-					UpdateStrategy: ptr.To(gardencorev1beta1.ManualInPlaceUpdate),
-				},
-			}
 			empty    *extensionsv1alpha1.OperatingSystemConfig
 			expected []*extensionsv1alpha1.OperatingSystemConfig
 		)
 
-		computeExpectedOperatingSystemConfigs := func(sshAccessEnabled, inPlaceUpdate bool) []*extensionsv1alpha1.OperatingSystemConfig {
+		computeExpectedOperatingSystemConfigs := func(sshAccessEnabled bool, workers []gardencorev1beta1.Worker, inPlaceUpdate bool) []*extensionsv1alpha1.OperatingSystemConfig {
 			w := workers
-			if inPlaceUpdate {
-				w = inPlaceUpdateWorkers
-			}
 
 			expected := make([]*extensionsv1alpha1.OperatingSystemConfig, 0, 2*len(w))
 			for _, worker := range w {
@@ -415,6 +342,82 @@ var _ = Describe("OperatingSystemConfig", func() {
 			Expect(fakeClient.Create(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "ca", Namespace: namespace}})).To(Succeed())
 			Expect(fakeClient.Create(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "ca-kubelet", Namespace: namespace}})).To(Succeed())
 
+			workers = []gardencorev1beta1.Worker{
+				{
+					Name: worker1Name,
+					Machine: gardencorev1beta1.Machine{
+						Architecture: ptr.To(v1beta1constants.ArchitectureAMD64),
+						Image: &gardencorev1beta1.ShootMachineImage{
+							Name:           "type1",
+							Version:        ptr.To("12.34"),
+							ProviderConfig: &runtime.RawExtension{Raw: []byte(`{"foo":"bar"}`)},
+						},
+					},
+					KubeletDataVolumeName: &kubeletDataVolumeName,
+					ControlPlane:          &gardencorev1beta1.WorkerControlPlane{},
+				},
+				{
+					Name: worker2Name,
+					Machine: gardencorev1beta1.Machine{
+						Architecture: ptr.To(v1beta1constants.ArchitectureAMD64),
+						Image: &gardencorev1beta1.ShootMachineImage{
+							Name:    "type2",
+							Version: ptr.To("12.34"),
+						},
+					},
+					CRI: &gardencorev1beta1.CRI{
+						Name: gardencorev1beta1.CRINameContainerD,
+					},
+					KubeletDataVolumeName: &kubeletDataVolumeName,
+					Kubernetes: &gardencorev1beta1.WorkerKubernetes{
+						Version: &workerKubernetesVersion,
+					},
+				},
+			}
+			inPlaceUpdateWorkers = []gardencorev1beta1.Worker{
+				{
+					Name: worker1Name + "-in-place",
+					Machine: gardencorev1beta1.Machine{
+						Architecture: ptr.To(v1beta1constants.ArchitectureAMD64),
+						Image: &gardencorev1beta1.ShootMachineImage{
+							Name:           "type1",
+							Version:        ptr.To("12.34"),
+							ProviderConfig: &runtime.RawExtension{Raw: []byte(`{"foo":"bar"}`)},
+						},
+					},
+					KubeletDataVolumeName: &kubeletDataVolumeName,
+					Kubernetes: &gardencorev1beta1.WorkerKubernetes{
+						Kubelet: &gardencorev1beta1.KubeletConfig{
+							KubeReserved: &gardencorev1beta1.KubeletConfigReserved{
+								CPU:    ptr.To(resource.MustParse("100m")),
+								Memory: ptr.To(resource.MustParse("100Mi")),
+							},
+						},
+					},
+
+					UpdateStrategy: ptr.To(gardencorev1beta1.AutoInPlaceUpdate),
+				},
+				{
+					Name: worker2Name + "-in-place",
+					Machine: gardencorev1beta1.Machine{
+						Architecture: ptr.To(v1beta1constants.ArchitectureAMD64),
+						Image: &gardencorev1beta1.ShootMachineImage{
+							Name:    "type2",
+							Version: ptr.To("12.34"),
+						},
+					},
+					CRI: &gardencorev1beta1.CRI{
+						Name: gardencorev1beta1.CRINameContainerD,
+					},
+					KubeletDataVolumeName: &kubeletDataVolumeName,
+					Kubernetes: &gardencorev1beta1.WorkerKubernetes{
+						Version: &workerKubernetesVersion,
+						Kubelet: &gardencorev1beta1.KubeletConfig{},
+					},
+					UpdateStrategy: ptr.To(gardencorev1beta1.ManualInPlaceUpdate),
+				},
+			}
+
 			values = &Values{
 				Namespace:         namespace,
 				Workers:           workers,
@@ -462,7 +465,7 @@ var _ = Describe("OperatingSystemConfig", func() {
 `)},
 			}
 
-			expected = computeExpectedOperatingSystemConfigs(false, false)
+			expected = computeExpectedOperatingSystemConfigs(false, workers, false)
 			defaultDepWaiter = New(log, c, sm, values, time.Millisecond, 250*time.Millisecond, 500*time.Millisecond)
 		})
 
@@ -658,7 +661,7 @@ var _ = Describe("OperatingSystemConfig", func() {
 
 				Expect(defaultDepWaiter.Deploy(ctx)).To(Succeed())
 
-				for _, e := range computeExpectedOperatingSystemConfigs(false, false) {
+				for _, e := range computeExpectedOperatingSystemConfigs(false, workers, false) {
 					actual := &extensionsv1alpha1.OperatingSystemConfig{}
 					Expect(c.Get(ctx, client.ObjectKey{Name: e.Name, Namespace: e.Namespace}, actual)).To(Succeed())
 
@@ -681,7 +684,7 @@ var _ = Describe("OperatingSystemConfig", func() {
 
 				Expect(defaultDepWaiter.Deploy(ctx)).To(Succeed())
 
-				for _, e := range computeExpectedOperatingSystemConfigs(true, false) {
+				for _, e := range computeExpectedOperatingSystemConfigs(true, workers, false) {
 					actual := &extensionsv1alpha1.OperatingSystemConfig{}
 					Expect(c.Get(ctx, client.ObjectKey{Name: e.Name, Namespace: e.Namespace}, actual)).To(Succeed())
 
@@ -737,7 +740,7 @@ var _ = Describe("OperatingSystemConfig", func() {
 
 					Expect(defaultDepWaiter.Deploy(ctx)).To(Succeed())
 
-					for _, e := range computeExpectedOperatingSystemConfigs(true, true) {
+					for _, e := range computeExpectedOperatingSystemConfigs(true, inPlaceUpdateWorkers, true) {
 						actual := &extensionsv1alpha1.OperatingSystemConfig{}
 						Expect(c.Get(ctx, client.ObjectKey{Name: e.Name, Namespace: e.Namespace}, actual)).To(Succeed())
 
@@ -788,7 +791,7 @@ var _ = Describe("OperatingSystemConfig", func() {
 				}
 			})
 
-			Context("CABundle generation", func() {
+			Context("CA bundle computation", func() {
 				var getROOTcertsFileContent = func(osc *extensionsv1alpha1.OperatingSystemConfig) *string {
 					for _, f := range osc.Spec.Files {
 						if f.Path == "/var/lib/ca-certificates-local/ROOTcerts.crt" {
@@ -800,48 +803,50 @@ var _ = Describe("OperatingSystemConfig", func() {
 					return nil
 				}
 
-				BeforeEach(func() {
-					values.OriginalValues.CABundle = "foo"
-					values.Workers[0].CABundle = ptr.To("bar")
-					values.Workers[1].CABundle = ptr.To("baz")
-				})
 				JustBeforeEach(func() {
 					Expect(defaultDepWaiter.Deploy(ctx)).To(Succeed())
 				})
 
-				It("should successfully generate caBundle when Worker.CABundle is provided", func() {
-					for _, e := range expected {
-						if e.Spec.Purpose != extensionsv1alpha1.OperatingSystemConfigPurposeReconcile {
-							continue
-						}
-						actual := &extensionsv1alpha1.OperatingSystemConfig{}
-						Expect(c.Get(ctx, client.ObjectKey{Name: e.Name, Namespace: e.Namespace}, actual)).To(Succeed())
-						worker := actual.Labels["worker.gardener.cloud/pool"]
-						content := getROOTcertsFileContent(actual)
-						Expect(content).NotTo(BeNil())
-						if worker == "worker1" {
-							Expect(*content).To(Equal("foo\nbar"))
-						} else {
-							Expect(*content).To(Equal("foo\nbaz"))
-						}
-					}
-				})
-
-				When("only original values CABundle is set", func() {
-					BeforeEach(func() {
-						values.Workers[0].CABundle = nil
-						values.Workers[1].CABundle = nil
-					})
-					It("should contains only original CABundle value", func() {
+				When("worker CA bundle is not set", func() {
+					It("should contain only the provided CA bundle", func() {
 						for _, e := range expected {
 							if e.Spec.Purpose != extensionsv1alpha1.OperatingSystemConfigPurposeReconcile {
 								continue
 							}
+
 							actual := &extensionsv1alpha1.OperatingSystemConfig{}
 							Expect(c.Get(ctx, client.ObjectKey{Name: e.Name, Namespace: e.Namespace}, actual)).To(Succeed())
+
 							content := getROOTcertsFileContent(actual)
 							Expect(content).NotTo(BeNil())
-							Expect(*content).To(Equal("foo"))
+							Expect(*content).To(Equal(caBundle))
+						}
+					})
+				})
+
+				When("worker CA bundle is set", func() {
+					BeforeEach(func() {
+						values.Workers[0].CABundle = ptr.To("foo")
+						values.Workers[1].CABundle = ptr.To("bar")
+					})
+
+					It("should append worker CA bundle to the CA bundle", func() {
+						for _, e := range expected {
+							if e.Spec.Purpose != extensionsv1alpha1.OperatingSystemConfigPurposeReconcile {
+								continue
+							}
+
+							actual := &extensionsv1alpha1.OperatingSystemConfig{}
+							Expect(c.Get(ctx, client.ObjectKey{Name: e.Name, Namespace: e.Namespace}, actual)).To(Succeed())
+							worker := actual.Labels["worker.gardener.cloud/pool"]
+
+							content := getROOTcertsFileContent(actual)
+							Expect(content).NotTo(BeNil())
+							if worker == "worker1" {
+								Expect(*content).To(Equal(fmt.Sprintf("%s\n%s", caBundle, "foo")))
+							} else {
+								Expect(*content).To(Equal(fmt.Sprintf("%s\n%s", caBundle, "bar")))
+							}
 						}
 					})
 				})

--- a/pkg/gardenlet/operation/botanist/operatingsystemconfig.go
+++ b/pkg/gardenlet/operation/botanist/operatingsystemconfig.go
@@ -92,7 +92,6 @@ func (b *Botanist) DeployOperatingSystemConfig(ctx context.Context) error {
 	if !found {
 		return fmt.Errorf("secret %q not found", v1beta1constants.SecretNameCACluster)
 	}
-
 	clusterCABundle, found := clusterCASecret.Data[secretsutils.DataKeyCertificateBundle]
 	if !found {
 		return fmt.Errorf("key %q not found in secret %q", secretsutils.DataKeyCertificateBundle, v1beta1constants.SecretNameCACluster)

